osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DeviceInstallationServiceTest.java

```diff
@@ -0,0 +1,276 @@
+		when(device.getIpAddress()).thenReturn("testIp");
+		when(this.deviceDomainService.searchActiveDevice("testDevice", ComponentType.DOMAIN_CORE)).thenReturn(device);
+
+		this.deviceInstallationService.getStatus("testOrganisation", "testDevice", "testUid", "testMessageType", 1);
+
+		verify(this.osgpCoreRequestMessageSender).send(this.argumentReqM.capture(), this.argumentStringOne.capture(),
+				this.argumentInt.capture(), this.argumentStringTwo.capture());
+
+		assertThat(this.argumentReqM.getValue()).usingRecursiveComparison().isEqualTo(new RequestMessage(
+				"testUid",
+				"testOrganisation",
+				"testDevice", null));
+		assertThat(this.argumentStringOne.getValue()).isEqualTo("GET_LIGHT_SENSOR_STATUS");
+		assertThat(this.argumentInt.getValue()).isEqualTo(1);
+		assertThat(this.argumentStringTwo.getValue()).isEqualTo("testIp");
+	}
+
+	@Test
+	public void testHandleGetStatusResponseNotOk() {
+		this.deviceInstallationService.handleGetStatusResponse(null, new CorrelationIds("testOrganisation", "testDevice", "testUid"),
+				"testMessageType", 1, ResponseMessageResultType.NOT_OK, null);
+
+		verify(this.webServiceResponseMessageSender).send(this.argumentResM.capture());
+		assertThat(this.argumentResM.getValue()).usingRecursiveComparison().isEqualTo(ResponseMessage.newResponseMessageBuilder()
+				.withIds(new CorrelationIds("testOrganisation", "testDevice", "testUid"))
+				.withResult(ResponseMessageResultType.NOT_OK).withOsgpException(null)
+				.withDataObject(null).withMessagePriority(1).build());
+	}
+
+	@Test
+	public void testHandleGetStatusResponseOkLmdStatusNotNull() throws FunctionalException, ValidationException {
+		final TariffValue editedTariffValue = new TariffValue();
+		editedTariffValue.setHigh(true);
+		editedTariffValue.setIndex(10);
+		final DeviceStatusMapped deviceStatus = new DeviceStatusMapped(null,
+				Arrays.asList(new LightValue(0, true, 50), new LightValue(1, true, 75), new LightValue(2, false, 0)),
+				LinkType.ETHERNET, LinkType.GPRS, LightType.ONE_TO_TEN_VOLT,0
+		);
+		when(this.domainCoreMapper.map(null, DeviceStatus.class)).thenReturn(deviceStatus);
+
+		final Device mockedDevice = Mockito.mock(Device.class);
+		when(mockedDevice.getDeviceType()).thenReturn(LightMeasurementDevice.LMD_TYPE);
+		when(this.deviceDomainService.searchDevice("testDevice")).thenReturn(mockedDevice);
+
+		this.deviceInstallationService.handleGetStatusResponse(null, new CorrelationIds("testOrganisation", "testDevice", "testUid"),
+				"testMessageType", 1, ResponseMessageResultType.OK, null);
+
+		verify(this.webServiceResponseMessageSender).send(this.argumentResM.capture());
+		assertThat(this.argumentResM.getValue()).usingRecursiveComparison().isEqualTo(ResponseMessage.newResponseMessageBuilder()
+				.withIds(new CorrelationIds("testOrganisation", "testDevice", "testUid"))
+				.withResult(ResponseMessageResultType.OK).withOsgpException(null)
+				.withDataObject(deviceStatus)
+				.withMessagePriority(1)
+				.build());
```

Comment:
Refactor the building of a response message to a method and use that method in other tests as well